### DISCUSSION TITLE
How to increase the rateLimit? (not the auth User rateLimit)

### DISCUSSION THREAD

**Discussion by gregpavl** [verdaccio/verdaccio#3428] [Q&A] [answered]
Based on the https://github.com/verdaccio/verdaccio/discussions/3427 (as an additional question/clarification)

Is there any way to increase the `rateLimit` of the incoming requests in `verdaccio`? 

I read from your `docs` that there is a configuration only auth user rateLimit (https://verdaccio.org/docs/configuration/#user-rate-limit) endpoints, however, this configuration does not include the `yarn/npm install` requests, but only the login & auths.

Is this assumption correct?

If yes, how can we configure these rateLimits, and which is the default request/sec ?

**Comment by gregpavl** [comment, marked as answer]
cc @juanpicado

**Reply by juanpicado** [reply]
yes, is intended, for the v5.x only includes auth security related endpoints https://github.com/verdaccio/verdaccio/pull/2800. For the next major should be in all endpoints. I'm not investing feature time on 5.x at this point but if anyone wants I'd be happy to review it and merge it if follow these recommendations:
- disabled by default (why, because endpoints are intensive, limit would be hard to set by default since I have no metrics how big should be)
- configuration example (this is just a suggestion) separated for the user.

User endpoints related (current one):

```
userRateLimit:
  windowMs: 50000 <- (minutes * 60 * 1000)
  max: 1000 (number of request peer windowMs)
````
could be to avoid conflicts:
```
apiRateLimit:
  windowMs: 50000 <- (minutes * 60 * 1000)
  max: 1000 (number of request peer windowMs)
````

For the next major could be different, so open to suggestions.
